test/ably/restpresence_test.py

```diff
@@ -21,18 +21,24 @@
 @six.add_metaclass(VaryByProtocolTestsMetaclass)
 class TestPresence(BaseTestCase):
 
+    @classmethod
+    def setUpClass(cls):
+        cls.ably = AblyRest(test_vars["keys"][0]["key_str"],
+                            rest_host=test_vars["host"],
```

Comment:
Are we not supporting `environment` ? See https://docs.ably.io/client-lib-development-guide/features/#RSC11

Reply:
It's supported by the library, see https://github.com/ably/ably-python/blob/develop/test/ably/restinit_test.py#L77

In the test suite the host is defined here https://github.com/ably/ably-python/blob/develop/test/ably/restsetup.py#L25 ; reading that code I think we're using "sandbox-rest.ably.io"

The code has been like that for a looong time. There's an issue to refactor this code, see #109 (refactoring proposed by @funkyboy ). What would you like to change exactly?

Reply:
> What would you like to change exactly?

There should be a way to run the tests against a non-sandbox environment by specifying only a single `environment` option. For example, with most or all the other libraries you can do:

`export ABLY_ENV=local; <run tests>`

The environment you specify is passed as the `environment` option each time the tests instance the library, but it is also used to determine the endpoint used to call `/apps` to create the test fixtures.

Reply:
Right now we can set `ABLY_HOST`, but no `ABLY_ENV`, see https://github.com/ably/ably-python/blob/develop/test/ably/restsetup.py#L20

I will add `ABLY_ENV`

Reply:
Do you think I should remove `ABLY_HOST`, or keep both?

Reply:
I think removing it is best. Thanks

Reply:
I've refactored the tests ( #109 ) so introducing `environment` will require only to change 1 file.

Now the tests are green.

Had some trouble, fixed it changing the order the tests are run, this may have fixed the random failures as well ( #122 ) ; something seems wrong in `reststats_test.py`